### PR TITLE
Add recipe for magpylib

### PR DESCRIPTION
**Pull request by OrtnerMichael**
Hello! This recipe is for magpylib, a Python 3.6+ API to calculate static electromagnetic field from fundamental types of permanent magnets, current distributions, and more. 

This has been in development for a while and only recently we've gone open-source :) 

I couldn't get pytest to find the unit tests within the docker due to our structure, but the minimal import test worked locally at least. Let us know if this is a problem.

Our PyPi and Github repository pages:

https://github.com/magpylib/magpylib
https://pypi.org/project/magpylib/

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages 
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/magpylib/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+{% set version = "1.0.2b0" %}
+{% set pypiSourceChecksum = 'fde2f8641a265461a9eb50259c5fccdf82822b57a441e3eae7a99f2861f3022e' %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ pypiSourceChecksum }}
```
Comment (chrisburr) [inline recipes/magpylib/meta.yaml]: Putting the checksum here will make it easier for the update bot:
```suggestion
  sha256: fde2f8641a265461a9eb50259c5fccdf82822b57a441e3eae7a99f2861f3022e
```

File: recipes/magpylib/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+{% set name = "magpylib" %}
+{% set version = "1.0.2b0" %}
+{% set pypiSourceChecksum = 'fde2f8641a265461a9eb50259c5fccdf82822b57a441e3eae7a99f2861f3022e' %}
```
Comment (chrisburr) [inline recipes/magpylib/meta.yaml]:
```suggestion
```

File: recipes/magpylib/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - python>=3.6
+    - pip
+  run:
+    - python
+    - matplotlib>=3.1
```
Comment (chrisburr) [inline recipes/magpylib/meta.yaml]: You need a space here like this:
```suggestion
    - matplotlib >=3.1
```

File: recipes/magpylib/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    - numpy>=1.16
```
Comment (chrisburr) [inline recipes/magpylib/meta.yaml]:
```suggestion
    - numpy >=1.16
```

File: recipes/magpylib/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+test:
+  imports:
+    - {{ name }}
+
+about:
+  home: http://github.com/magpylib/magpylib
+  license: AGPL-3.0
```
Comment (chrisburr) [inline recipes/magpylib/meta.yaml]: According to the `setup.py`:
```suggestion
  license: AGPL-3.0-or-later
```

File: recipes/magpylib/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ pypiSourceChecksum }}
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
Comment (chrisburr) [inline recipes/magpylib/meta.yaml]: Do you need a compiler? At a glance this looks like it's pure Python

File: recipes/magpylib/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ pypiSourceChecksum }}
+
+build:
+  number: 0
```
Comment (chrisburr) [inline recipes/magpylib/meta.yaml]: If you don't need a compiler:
```suggestion
  number: 0
  noarch: python
```
And constrain both the `host` and `run` dependencies to be `python >=3.6`.

File: recipes/magpylib/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    group several sources into a Collection system and to calculate the
+    resulting fields of such systems. The systems may also be quickly visualized
+    with a matplotlib based 3D plot method.
+  doc_url: https://magpylib.readthedocs.io/en/latest/
+  dev_url: https://github.com/magpylib/magpylib
+
+extra:
+  recipe-maintainers:
+    - OrtnerMichael
+    - lucasgcb
```
Comment (chrisburr) [inline recipes/magpylib/meta.yaml]: @lucasgcb Are you happy to be listed as a maintainer of this recipe?

Comment (OrtnerMichael) [reply recipes/magpylib/meta.yaml]: Yes, I am.

Comment (lucasgcb) [reply recipes/magpylib/meta.yaml]: Yes, I am.

File: recipes/magpylib/meta.yaml (hunk 6 above)
Comment (lucasgcb) [reply recipes/magpylib/meta.yaml]: Not right now, nope. We do plan on adding optimal extensions in the future though.

Comment (chrisburr) [reply recipes/magpylib/meta.yaml]: When that happens you should replace `noarch: python` with `skip: true  # [py<36]` and remove the version constraint on Python in `host` and `run`.

**Comment by jakirkham** [comment]
I'm not sure why, but [conversion of feedstocks stopped working]( https://travis-ci.org/conda-forge/staged-recipes/builds/538846202 ) after this was merged (maybe coincidental?). @chrisburr, would you have an opportunity to look at what is going on?

**Comment by chrisburr** [comment]
That error is caused by something unrelated to this PR. I can't fix it but I've let core know on Gitter so I'm sure it'll be fixed soon.

**Comment by jakirkham** [comment]
Works for me. Thanks!

**Pull request merged**